libtenzir/src/tql2/set.cpp

```diff
@@ -302,6 +302,13 @@ auto assign(const ast::selector& left, series right, const table_slice& input,
       auto result = std::vector<table_slice>{};
       result.push_back(assign(left, std::move(right), input, dh, position));
       return result;
+    },
+    [&](const ast::dollar_var& left) {
+      // TODO: I hope throwing is okay.
+      diagnostic::error("cannot assign to `{}` constant2", left.id.name)
+        .primary(left)
+        .throw_();
+      return std::vector<table_slice>{};
```

Comment:
I would want to avoid throwing here. Shouldn't this be unreachable if we did our previous checks correctly?